api/v1alpha1/authorization_types.go

```diff
@@ -76,12 +81,22 @@ type Principal struct {
 //
 // +kubebuilder:validation:XValidation:rule="(has(self.claims) || has(self.scopes))",message="at least one of claims or scopes must be specified"
 type JWTPrincipal struct {
+	// Provider is the name of the JWT provider that used to verify the JWT token.
+	// The provider must be configured in the JWT authentication.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Provider string `json:"provider"`
```

Comment:
can we do without this or is this needed ?

Reply:
It's a must as the JWT auth filter stores the extracted JWT claims under its namespace, and we need this to correlate the JWT principal with the JWT claims in the metadata.

```yaml
   envoy.filters.http.jwt_authn:
      jwt_provider_name:
         scopes: "read write"
         name: "John"
         sub: "John"
```